rdtools/filtering.py

```diff
@@ -102,25 +185,325 @@ def clip_filter(power_ac, quantile=0.98):
+        raise ValueError(
+            "Variable mounting_type must be string 'Tracking' or 'Fixed'.")
+    # Get the names of the series and the datetime index
+    column_name = power_ac.name
+    column_name = 'value'
+    power_ac = power_ac.rename(column_name)
+    index_name = power_ac.index.name
+    if index_name is None:
+        index_name = 'datetime'
+        power_ac = power_ac.rename_axis(index_name)
+    return power_ac, power_ac.index.name
+
 
+def _calculate_max_derivative(power_ac, roll_periods):
+    """
+    This function calculates the maximum derivative over a rolling
+    time period for an AC power time series. A pandas series of
+    the rolling derivative is returned.
+    
     Parameters
     ----------
-    poa_global_measured : pd.Series
-        Plane of array irradiance based on measurments
-    poa_global_clearsky : pd.Series
-        Plane of array irradiance based on a clear sky model
-    threshold : float, default 0.15
-        threshold for filter
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    roll_periods: Int
+        Number of readings to calculate the rolling maximum derivative on.
 
     Returns
     -------
     pd.Series
-        Boolean Series of whether the clear-sky index is within the threshold
-        around 1.
+        Time series of the rolling maximum derivative.
+    """
+    # Calculate the maximum value over a forward-rolling window
+    max_roll = power_ac.iloc[::-1].rolling(roll_periods).max()
+    max_roll = max_roll.reindex(power_ac.index)
+    # Calculate the minimum value over a forward-rolling window
+    min_roll = power_ac.iloc[::-1].rolling(roll_periods).min()
+    min_roll = min_roll.reindex(power_ac.index)
+    # Calculate the maximum derivative within the foward-rolling window
+    derivative_max = (max_roll - min_roll) / ((max_roll + min_roll) / 2) * 100
+    return derivative_max
+
+
+def logic_clip_filter(power_ac,
+                      mounting_type='Fixed',
+                      derivative_cutoff=0.2,
+                      roll_periods=None):
     '''
+    This filter is a logic-based filter that is used to filter out
+    clipping periods in AC power and AC energy time series.
+    The AC power or energy time series is filtered based on the
+    maximum derivative over a rolling window, as compared to a user-set
+    derivate_cutoff (default set to 0.2).  The size of the
+    rolling window is increased when the system is a tracked system.
+    
+    Parameters
+    ----------
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    mounting_type: string, default 'Fixed'
+        String representing the mounting configuration associated with the
+        AC power/energy time series. Can either be "Fixed" or "Tracking".
+        Default set to 'Fixed'.
+    derivative_cutoff : float, default 0.2
```

Comment:
I wonder if we can come up with a more accurate term than derivative. I think we're calculating something more like a "relative range" but I'm not sure if there is a better more technically correct term. "Relative variation" or something maybe? 

Also, I think it's a percentage, is that right? We should specify.

Reply:
Some similar commentary here: https://github.com/NREL/rdtools/pull/200#discussion_r591865932

Reply:
I still think we need to eliminate "derivative" in favor of something like "rolling range" or similar since we're not strictly calculating a derivative